packages/data-context/src/sources/UtilDataSource.ts

```diff
@@ -3,6 +3,16 @@ import crypto from 'crypto'
 import fetch from 'cross-fetch'
 import type { DataContext } from '../DataContext'
 
+// Require rather than import since data-context is stricter than network and there are a fair amount of errors in agent.
+const { agent } = require('@packages/network')
+
+const proxiedFetch = function (this: any, input: RequestInfo, init?: RequestInit) {
```

Comment:
Can't we just use a narrow function to capture the context?

```ts
const proxiedFetch = (input: RequestInfo, init?: RequestInit) => fetch(input, { agent, ...init })
```

Reply:
Changed here: [`71933ef` (#20107)](https://github.com/cypress-io/cypress/pull/20107/commits/71933ef5a6a332e30348bbead924caf4749061fc)